src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -31,25 +36,39 @@ public boolean isConsistent(Supplier<Connection> replicaSupplier) {
 
     public static final class Builder {
         private ReplicaConsistency replicaConsistency;
-        private AuroraConnectionDetails auroraConnectionDetails;
+        private ConnectionPoolPerUrlProvider connectionPoolPerUrlProvider;
 
+        public Builder replicaConsistency(ReplicaConsistency replicaConsistency) {
+            this.replicaConsistency = replicaConsistency;
+            return this;
+        }
+
+        /**
+         * @return new instance of this class
+         * @deprecated use {@link AuroraMultiReplicaConsistency#builder()} instead.
+         */
+        @Deprecated
         public static Builder anAuroraMultiReplicaConsistencyBuilder() {
             return new Builder();
         }
 
-        public Builder replicaConsistency(ReplicaConsistency replicaConsistency) {
-            this.replicaConsistency = replicaConsistency;
+        public Builder connectionPoolPerUrlProvider(ConnectionPoolPerUrlProvider connectionPoolPerUrlProvider) {
```

Comment:
IMO we should rename type or at least parameter. Shouldn't client know that this have to be connection pool provider for read replicas? Configuration between main / replica connection pool may differ, so we have to suggest that we will be creating connections to replicas.

Reply:
will rename to -PerReplica-